cnxarchive/utils/ident_hash.py

```diff
@@ -82,3 +81,186 @@ def join_ident_hash(id, version):
     if version:
         join_args.append(version)
     return HASH_CHAR.join(join_args)
+
+
+class CNXHash(uuid.UUID):
```

Comment:
The UUID class is already used throughout our source code and it has its own datatype in postgresql.  Plus, the UUID class can already accept strings, hex digits and bytes.  Since I wanted to be able to calculate a short id no matter where a uuid was used or what form it was in.  Looking it over again, I would have liked to make it more 'python-y'.